File: 剑指offer代码自编/3_search_in_matrix.cpp
bool find(int* matrix, int cols, int rows, int k)
{
	bool found = True;
	if(matrix != NULL && cols > 0 && rows > 0)
	{
		int row = 0;
		int col = cols - 1;
		while(row < rows && col >= 0)
		{
			if(matrix[row * cols + col] == k)
				break;
			else if(matrix[row * cols + col] > k)
				-- col;
			else
				++ row;
		}
	}
	return found;
}

File: 剑指offer代码自编/3_search_in_matrix.py
def find(a, number):
	found = True
	if a == None or a == [[]] or a = []:
		return False
	rows = len(a)
	cols = len(a[0])
	row = 0
	col = cols - 1
	while(row < rows and col >= 0):
		if a[row][col] == number:
			return found
			break
		elif a[row][col] < number:
			row += 1
		else:
			col -= 1
	return found
